task3/UI/script_task7.js

```diff
@@ -0,0 +1,527 @@
+class AdCollection {
+    constructor(adList) {
+        if (adList) {
+            this._adList = adList.slice();
+        } else {
+            this._adList = [];
+        }
+    }
+
+    add(adItem) {
+        if (AdCollection._validate(adItem) && !this.get(adItem.id)) {
+            this._adList.push(adItem);
+            return true;
+        }
+        return false;
+    }
+
+    addAll(adList) {
+        let notValidPosts = [];
+        adList.forEach((post) => {
+            if (AdCollection._validate(post) && this.get(post.id)) {
```

Comment:
I think here should be `if (AdCollection._validate(post) && !this.get(post.id)) {`

Reply:
Thank you! 
I fixed all thing that you commented, can i merge this task?

Reply:
You can, but first please fix a couple of new defects of `edit` method 🙂